OMCompiler/SimulationRuntime/fmi/export/openmodelica/fmu2_model_interface.c.inc

```diff
@@ -647,8 +647,8 @@ fmi2Status fmi2ExitInitializationMode(fmi2Component c)
     }
   }
 
-  /* TODO: Simulation stop time is needed to calculate the sample events beforehand. */
-  initSample(comp->fmuData, comp->threadData, comp->fmuData->localData[0]->timeValue, 100 /*should be stopTime*/);
+  /* use defined stopTime, if stopTimeDefined is given to calculate the sample events beforehand. */
+  initSample(comp->fmuData, comp->threadData, comp->fmuData->localData[0]->timeValue, comp->stopTimeDefined ? comp->stopTime : 100);
```

Comment:
It's strange to be mixing integer numbers with floats... What is 100? Is that the number of slots to be used for allocation? Or the default simulation time?

Reply:
The initiSample is defined as:
`void initSample(DATA *data, threadData_t *threadData, double start, double stop);`
so it doesn't really matter if you send it 100.0 or 100. 
That is some sort of default for stopTime, if stopTime is not defined.